crypto/modes/gcm128.c

```diff
@@ -369,7 +369,7 @@ void gcm_gmult_4bit_x86(u64 Xi[2], const u128 Htable[16]);
 void gcm_ghash_4bit_x86(u64 Xi[2], const u128 Htable[16], const u8 *inp,
                         size_t len);
 #  endif
-# elif defined(__arm__) || defined(__arm) || defined(__aarch64__)
+# elif defined(__arm__) || defined(__arm) || defined(__aarch64__) || defined(_M_ARM64)
```

Comment:
What about `_M_ARM64EC`? (multiple places)

Reply:
_M_ARM64EC check could be a good idea which deserves an additional PR, testing and benchmarking.

Reply:
If it's just going to be ` || defined(_M_ARM64EC)` in a few places, might be better in this one?

Out of interest, what system(s) are you testing on (hardware-wise)?

Reply:
_M_ARM64EC is not defined there. I expect it will appear when Arm64EC architecture will be selected.
It is tested on an Azure Arm64 VM with Windows 11 Arm64 and Ampere(R) Altra(R) Processor, 2999 Mhz, 2 Cores, 2 Logical Processors

Reply:
I am not sure what is current status of Arm64EC support. Unfortunately it looks like we cannot allocate resources for this work now. 
Arm64 Windows optimization has been already supported by OpenSSL, however it had some bottleneck.
The purpose of this work was to identify Arm64 performance bottleneck on Arm64 Windows and resolve it.

The bottleneck has been identified and fixed in this PR. It allowed to decrease performance difference for P99 benchmark tests Arm64 vs AArch64 on the same hardware scale from 25 times slower than AArch64 to 2.3 times slower.

Reply:
I don't want to hold this up, so let's worry about Arm64EC later then